src/opt/ImproveMemoryOps.cpp

```diff
@@ -918,137 +918,97 @@ static llvm::CallInst *lGSToGSBaseOffsets(llvm::CallInst *callInst) {
     return newCall;
 }
 
+static std::unordered_map<std::string, int> checksDecompositionOfArgs = {
```

Comment:
`std::unordered_map<std::string, int>` - I'm scratching my head why is it `std::string`? Why not a pointer, because that's what the identifier is and that's more efficient.

But does this really mean that we need a typedef for a builtin type?

Reply:
The problem is in equality/hashing. With `string`, equality checks the content of the string. With pointers, it checks pointer value (i.e., the address of the string). Even if two `char*` pointers point to identical strings, they are considered different keys if they point to different addresses.

Reply:
But the c-string identifying the builtin is unique. At least with the redesign the builtin is assumed to be declared once and used only by the handle. So I don't see an issue here.

I'm fine with the string, but it's kinda unexpected pivot after the builtins redesign. I would say it's up to you if you'd like keep it this way or introduce the type and use it as a key for the map.

Reply:
> But the c-string identifying the builtin is unique. 

They are but loaded LLVM module has no connection with them, so `call->GetParent()->GetName()` does not refer/contain a c-string from `builtin-decl.cpp`.